### ISSUE TITLE
WoA (Atomic) Site Revert: Revert process sending out email notifications for scheduled posts

### ISSUE THREAD

**Issue by retnonindya** [Automattic/wp-calypso#50881]
When a simple site has scheduled posts and transferred to Atomic environment, the scheduled posts are kept on the scheduled process -- despite the posts themselves already published when the website is in AT environment.

When the revert occurs, the Simple site starts running its normal jobs and checks for scheduled posts that are upcoming or have missed their schedule. Since it has scheduled some, it starts publishing them.

P2 (discussion and reports):

- p9F6qB-6x5-p2 
- p9F6qB-6iZ-p2#comment-35983 
- p9GEWz-za-p2

This is far from ideal. Is there any way we can ensure the scheduled posts are not getting published and sending out notifications when site is reverted?

**Comment by dsas** [comment]
This is happening because there are scheduled posts in the simple DB. These are copied across to the atomic db and the scheduled posts published (and any notifications sent) on atomic when the posts reach their publication date. When the site is reverted back to simple infra, the simple db is activated again with it's pre-atomic content, and so if the scheduled posts are now in the past they will now become published (and any notifications sent).

Ideas on how this could be addressed (haven't looked at the code yet):
 * If we know we're going to restore, then block a site from sending email notifications until the restore is complete
   * Doesn't solve the problem for when the user doesn't want to restore.
 * Don't send any notifications for posts that have a published date that is "a long time ago" 
   * Not sure what counts as a long time ago. A week? How far is the notification system ever likely to be behind?
   * Doesn't solve the problem entirely, though reduces the impact.
   * Risk it's too far reaching - not tied to atomic revert
   * Do people sometimes schedule posts to be in the distant past on purpose? How much do we worry about those people?
 * As part of the revert, mark as published any scheduled posts on simple that are in the past, so they don't get sent again.
   * This assumes they've been sent on atomic, but that seems safer than the current "assumption" of assuming they haven't.

**Comment by dsas** [comment]
Currently we're not offering the option to restore. During the self-initiated revert and restore process there is a blog sticker 'subscription_deactivation_locked' which could be used to inibit sending email notifications, however this isn't in place for automatic expiry revert and restore process.

**Comment by jordesign** [comment]
Noting this is still relevant and ocurring. Not regularly, but definitely a poor part of the product experience to improve.

**Comment by p-jackson** [comment]
@jordesign does the fact we have the lossless revert process now change things? It used to be reverting an atomic site meant going back to your old content, including the old post schedules.
Now that the lossless revert process attempts to bring the atomic content back to the simple database, doesn't that mean the published statuses can cover over too? If the the email was sent while the site was atomic, then the simple database should eventually reflect the fact that the email for a given post has already been sent?

**Comment by supernovia** [comment]
@p-jackson does the lossless process run without HE intervention, if a site's plan expires for example?

**Comment by dsas** [comment]
@supernovia yes the automated downgrade to simple should do a lossless revert.

FWIW I don't remember this and I'd only been at Automattic a couple of months by that point so it might be worth verifying what I said is currently true.

**Assuming** I was right though, what I was describing happens during the lossless revert. The simple site becomes "active", and then there's a race between the previously scheduled posts going out and the import from atomic updating the post status to publish.

There's a comment here which suggests an approach to tackling this: pdgTfM-35-p2#comment-94

**Comment by jordesign** [comment]
> The simple site becomes "active", and then there's a race between the previously scheduled posts going out and the import from atomic updating the post status to publish.

This is the situation I would expect (especially from the perspective of doing manual downgrades). There _is_ a period between the revert to simple site, and then the content being imported/updated. So the concern is those scheduled posts would still go out in that period...

Although... we do make the site private at some point in that process. Which I think _should_ stop notifications from going out?

**Comment by supernovia** [comment]
I don't know whether we always go lossless, and I do know we make sites private. 

I'm not sure how best to test this, but here's what I'm trying
- I've just added a scheduled post on a test site (ID 866)
- then upgraded and changed to Atomic
- set plan to expire today
- deleted the scheduled post from Atomic

I'm not sure when it will auto revert, but when it does I'll check:
- whether the site is private
- whether the post is there
- whether emails went out

Setting to "In Triage" for now

**Comment by Robertght** [comment]
I just ran through the Business plan cancelation flow and this is what I experienced:

- After the downgrade, the recently published content(2 posts) was missing, but I waited for the lossless revert to complete.
- The site was private

Site: geGek-p2

@dsas @p-jackson do we know what team could further look into this or should we keep it for Groundskeeping?

**Comment by Robertght** [comment]
Looping in @Automattic/loop and assigning this to you for now.

**Comment by davemart-in** [comment]
@Robertght this one doesn't feel like a great fit for Loop either. Mind sharing what criteria you considered before assigning things to Loop? This feel more like something the Atomic team might want to look into.

**Comment by Robertght** [comment]
Thank you @davemart-in !

Sure! I was working of this: PdqkMK-1nV-p2

I'm not sure if we have a team working on this nor I'm not sure which is creating the actual issues, but I'd love to learn more about it.

**Comment by davemart-in** [comment]
Okay, thanks for clarifying. Please continue assigning things to us and we'll work through them issue by issue.
